### ISSUE TITLE
p-calendar: Screen reader will not read values in the calendar overlay

### ISSUE THREAD

**Issue by eloykrikken** [primefaces/primeng#11934]
### Describe the bug

Using NVDA screen reader and navigation via keyboard, the screen reader does not read the values of the datepicker correctly. NVDA will read out the column or row number instead.
Navigation seems to break because using up and down arrow is moving left and right in the datepicker. Visually I get no feedback what value is in focus, since highlighting stays on the default selected day. These navigation issues seem to only pop-up when I have NVDA active in the background. When closin NVDA navigation works correctly

### Environment

A webapp using PrimeNG with Serenity Template. NVDA as a screenreader.

### Angular version

9.0.4

### PrimeNG version

9.0.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

16.14.2

### Browser(s)

Chrome, Firefox

### Steps to reproduce the behavior

1. Go to a instance of the p-calendar
2. Open NVDA screen reader
3. Click to open the datepicker or navigate to it using the keyboard
4. Use TAB to select a date in the datepicker
5. Use the arrowkeys to select a different date
6. Listen to the screen reader. The values it reads are not what is displayed on screen and navigation does also not work properly

### Expected behavior

I expect that the screenreader will read the date number out lout and that I can see what date is currently in focus.
